vassal-app/src/main/java/VASSAL/tools/BackgroundTask.java

```diff
@@ -17,45 +17,12 @@
  */
 package VASSAL.tools;
 
-import javax.swing.SwingUtilities;
+import javax.swing.SwingWorker;
 
 /**
- * Utility task for starting a thread that performs one task,
- * {@link #doFirst}, then queues another another
- * task, {@link #doLater}, for the Event Handler thread to execute
- * This is basically a simple version of Sun's SwingWorker class.
- *
  * @deprecated Use {@link SwingWorker} now that we ship the JAR for it.
  */
 @Deprecated
```

Comment:
Why delete all the insides of this class but not the whole class? (Is something depending on the class, but none of it's members?)

Reply:
It's weird but yes it looks like it, else the class itself would appear in the unused-dependencies list, like other classes which don't have any dependents anymore.

Reply:
`BackgroundTask` is used by `CASL.VASL.VASLThread` in the VSQL module:

https://sourceforge.net/p/vassalengine/svn/HEAD/tree/VSQL-src/trunk/CASL/VASL/VASLThread.java

I think that's current? (@BrentEaston would know.) I don't see how that can possibly compile after gutting `BackgroundTask`, since it won't have a `start()` to call.

Reply:
We have a detected dependency on the class itself and it's constructor by VASLThread:
https://github.com/vassalengine/vassal/blob/6f8c7f08dc3801669b07c07195e6713a6c9762a1/vassal-app/src/test/resources/depreport/depreport.txt#L155

But not on the methods inside it, that's why they appear among unused dependencies: https://github.com/vassalengine/vassal/blob/6f8c7f08dc3801669b07c07195e6713a6c9762a1/vassal-app/src/test/resources/depreport/unused-dependencies.txt#L126

Either the module we did the dependency analysis for is a different version from the SVN trunk, or it's an issue with the analysis tool, maybe it cannot detect method calls on anonymous types?

Reply:
Either way, I reverted `BackgroundTask` to how it was before.